decoder.go

```diff
@@ -36,51 +34,46 @@ func (d *decoder) readBits(f field, outputLength uint8) (output []byte) {
 	}
 
 	// originPos: Original position of the first bit in the first byte
-	var originPos uint8 = 8 - d.bitCounter
+	originPos := 8 - d.bitCounter
 
 	// destPos: Destination position ( in the result ) of the first bit in the first byte
-	var destPos uint8 = f.BitSize % 8
+	destPos := f.BitSize % 8
 	if destPos == 0 {
 		destPos = 8
 	}
 
 	// numBytes: number of complete bytes to hold the result
-	var numBytes uint8 = f.BitSize / 8
+	numBytes := f.BitSize / 8
 
 	// numBits: number of remaining bits in the first non-complete byte of the result
-	var numBits uint8 = f.BitSize % 8
+	numBits := f.BitSize % 8
 
 	// number of positions we have to shift the bytes to get the result
-	var shift uint8 = (uint8(math.Abs(float64(originPos - destPos)))) % 8
```

Comment:
originPos and destPos are unsigned integers, so subtracting one from another will always result in a positive integer, so Abs is not necessary (or effective)